rsocket/statemachine/StreamsWriter.h

```diff
@@ -24,18 +19,12 @@ class StreamsWriter {
       Payload payload,
       bool TEMP_completed) = 0;
 
-  virtual void writeRequestN(StreamId streamId, uint32_t n) = 0;
-
-  virtual void
-  writePayload(StreamId streamId, Payload payload, bool complete) = 0;
+  virtual void writeRequestN(Frame_REQUEST_N&&) = 0;
```

Comment:
Let's simplify it further as renaming this method as `write`?
```
virtual void write(std::unique_ptr<folly::IOBuf> frame)
```
Why do we care about what do we write?

Reply:
See my other comment, but there's different behavior for writing different frames (e.g. `REQUEST_N` versus `REQUEST_STREAM`).